packages/compiler/src/formatter/print/printer.ts

```diff
@@ -1042,27 +1055,61 @@ function joinPropertiesInBlock(
   const doc: Doc[] = [regularLine];
   const propertyContainerNode = path.getValue();
 
+  const propct = (propertyContainerNode.properties ?? []).length;
+  const validct = (propertyContainerNode.validates ?? []).length;
+
   let newLineBeforeNextProp = false;
-  path.each((item, propertyIndex) => {
-    const isFirst = propertyIndex === 0;
-    const isLast = propertyIndex === propertyContainerNode.properties.length - 1;
-    const shouldWrapInNewLines = shouldWrapPropertyInNewLines(item as any, options);
-
-    if ((newLineBeforeNextProp || shouldWrapInNewLines) && !isFirst) {
-      doc.push(hardline);
-      newLineBeforeNextProp = false;
-    }
-    doc.push(print(item));
-    if (isLast) {
-      doc.push(ifBreak(separator));
-    } else {
-      doc.push(separator);
-      doc.push(regularLine);
-      if (shouldWrapInNewLines) {
-        newLineBeforeNextProp = true;
+  if (propct > 0) {
+    path.each((item, propertyIndex) => {
+      const isFirst = propertyIndex === 0;
+      const isLast = propertyIndex === propct - 1 && validct === 0;
+      const shouldWrapInNewLines = shouldWrapPropertyInNewLines(item as any, options);
+
+      if ((newLineBeforeNextProp || shouldWrapInNewLines) && !isFirst) {
+        doc.push(hardline);
+        newLineBeforeNextProp = false;
       }
-    }
-  }, "properties");
+      doc.push(print(item));
+      if (isLast) {
+        doc.push(ifBreak(separator));
+      } else {
+        doc.push(separator);
+        doc.push(regularLine);
+        if (shouldWrapInNewLines) {
+          newLineBeforeNextProp = true;
+        }
+      }
+    }, "properties");
+  }
+
+  if (propct > 0 && validct > 0) {
+    doc.push(hardline);
+    newLineBeforeNextProp = false;
+  }
+
+  if (validct > 0) {
+    path.each((item, propertyIndex) => {
+      const isFirst = propertyIndex === 0 && propct === 0;
+      const isLast = propertyIndex === validct - 1;
+      const shouldWrapInNewLines = shouldWrapPropertyInNewLines(item as any, options);
```

Comment:
Think that function should also be changed to take the validate node into account